packages/components/src/components/editor/condition-builder/ConditionBuilderContext.tsx

```diff
@@ -0,0 +1,155 @@
+import { createContext, useContext, useState, type ReactNode } from 'react';
+import type { Condition } from './Condition';
+import type { ConditionGroup } from './ConditionGroup';
+import type { ConditionMode } from './ConditionBuilder';
+import { Input } from '@/components/common/input/input';
+
+export type TypeOperator = {
+  'equal to': string;
+  'not equal to': string;
+  'is true': string;
+  'is false': string;
+  'is empty': string;
+  'is not empty': string;
+  'less than': string;
+  'greater than': string;
+  'less or equal to': string;
+  'greater or equal to': string;
+};
+
+export type ConditionType = keyof TypeOperator;
```

Comment:
Unfortunately, I haven't found a way to define a type that takes all the values from, for example, typesOptions (since the typesOptions object can't be accessed at the time the type is generated). So i just use the keyof TypeOperator. Also not quiet sure about the naming of TypeOperator/ConditionType and LogicOperator/LogicalOperator

Reply:
It is not possible to generate a type of something which is not constant or already an type
But why would you need that, just make a type of the static part of the type and use it to access the value for that key.

Reply:
not sure too... maybe:

TypeOperator -> OperatorType (or just Operators)
ConditionType -> Operator
LogicOperator -> LogicOperators

I think I would rename the ConditionType, as this sais nothing right? it is an operator not a condition type...